taichi/transforms/check_out_of_bound.cpp

```diff
@@ -26,32 +26,44 @@ class CheckOutOfBound : public BasicStmtVisitor {
       return;
     TI_ASSERT(stmt->snodes.size() == 1);
     auto snode = stmt->snodes[0];
+    bool has_offset = !(snode->index_offsets.empty());
     auto new_stmts = VecStatement();
     auto zero = new_stmts.push_back<ConstStmt>(LaneAttribute<TypedConstant>(0));
     Stmt *result =
         new_stmts.push_back<ConstStmt>(LaneAttribute<TypedConstant>(true));
 
     std::string msg = "Accessing Tensor of Size [";
+    std::string offset_msg = "Offset [";
     std::vector<Stmt *> args;
     for (int i = 0; i < stmt->indices.size(); i++) {
-      auto check_zero = new_stmts.push_back<BinaryOpStmt>(
-          BinaryOpType::cmp_ge, stmt->indices[i], zero);
+      int offset_i = has_offset ? snode->index_offsets[i] : -1;
```

Comment:
nit: If this is `int offset_i = has_offset ? snode->index_offsets[i] : 0;`, we can reuse `zero` when some `offset_i == 0`, and `int upper_bound_i = has_offset ? offset_i + size_i : size_i;` will be unnecessary.

Reply:
Thanks for pointing out this!